From 360b67c22754349f23952071325aa4270e9211f6 Mon Sep 17 00:00:00 2001
From: Varun <vyaparla@habana.ai>
Date: Mon, 28 Feb 2022 11:38:16 +0200
Subject: [PATCH] [SW-64613] gaudi2: remove security strap check code

This change removes security enable strap check code
for gaudi2 arm. In gaudi2, security bypass strap is
used to enable arc boot flow.

Change-Id: If37c465d400a563dd7e8dd7c35cfa1210120c683
---

diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 69facc2..1a97856 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -414,18 +414,6 @@
 #define EFUSE_SECURITY_WORD_OFFSET		0x3A
 #define EFUSE_SECURITY_IMG_VALIDATION_BIT	BIT(10)
 
-static bool hl_is_security_strap_enabled(void)
-{
-	/*
-	 * TODO: SECURITY_BYPASS boot strap pin in gaudi2
-	 * is used for enabling/disabling arcboot flow (connected
-	 * to GPIO_47). Hence strap check to be confirmed with
-	 * system team if additional indicator like otp is needed
-	 * to make the solution robust like gaudi.
-	 */
-	return false;
-}
-
 static bool preboot_sec_efuse_is_set(u32 mask)
 {
 	u32 efuse_word;
@@ -439,15 +427,10 @@
 
 bool preboot_security_is_enabled(void)
 {
-	/* If security is enabled via boot strap pins */
-	if (hl_is_security_strap_enabled())
-		return true;
-
 	if (preboot_sec_efuse_is_set(EFUSE_SECURITY_ENABLE_BIT))
 		return true;
 
 	return preboot_fake_security_is_enabled();
-
 }
 
 bool preboot_img_validation_is_enabled(void)
